tests/test_aliases.py

```diff
@@ -0,0 +1,291 @@
+    """
+    data = dict(a=1, b=2)
+    client.post(f"/metadata/{guid1}", json=data).raise_for_status()
+    client.post(f"/metadata/{guid2}", json=data).raise_for_status()
+
+    client.post(
+        f"/metadata/{guid1}/aliases", json={"aliases": aliases1}
+    ).raise_for_status()
+
+    try:
+        response = client.post(f"/metadata/{guid2}/aliases", json={"aliases": aliases2})
+        assert str(response.status_code) == "409"
+
+        # ensure the original aliases data did not get modified
+        assert client.get(f"/metadata/{guid1}").json() == data
+        assert client.get(f"/metadata/{guid1}/aliases").json().get("aliases") == sorted(
+            aliases1
+        )
+        assert client.get(f"/metadata/{guid1}/aliases").json().get("guid") == guid1
+
+        for alias in aliases1:
+            # NOTE you have to percent encode aliases in urls
+            assert client.get(f"/metadata/{urllib.parse.quote(alias)}").json() == data
+    finally:
+        client.delete(f"/metadata/{guid1}").raise_for_status()
+        client.delete(f"/metadata/{guid2}").raise_for_status()
+
+
+@pytest.mark.parametrize(
+    "guid,aliases,updates,merge",
+    [
+        ("test_get_aliases", ["alias_a"], ["alias_a", "another_alias"], False),
+        ("test_get_aliases", ["alias_a"], ["alias_a", "another_alias"], True),
+        ("test_get_aliases", ["alias_a"], ["alias_a"], False),
+        ("test_get_aliases", ["alias_a"], ["alias_a"], True),
+        (
+            "test_get_aliases",
+            ["alias_a"],
+            ["new_alias", "alias_a", "another_alias"],
+            False,
+        ),
+        (
+            "test_get_aliases",
+            ["alias_a"],
+            ["new_alias", "alias_a", "another_alias"],
+            True,
+        ),
+        ("test_get_aliases", ["alias_a"], ["only_alias"], False),
+        ("test_get_aliases", ["alias_a"], ["only_alias"], True),
+        (
+            "test_get_aliases",
+            ["alias_a"],
+            ["duplicate_alias", "duplicate_alias"],
+            False,
+        ),
+        ("test_get_aliases", ["alias_a"], ["duplicate_alias", "duplicate_alias"], True),
+        ("test_get_aliases", ["alias_a"], [], False),
+        ("test_get_aliases", ["alias_a"], [], True),
+    ],
+)
+def test_update_already_created_aliases(guid, aliases, updates, merge, client):
+    """
+    Ensure successful response when trying to PUT an already existing record with
+    new aliases
+    """
+    data = dict(a=1, b=2)
+    client.post(f"/metadata/{guid}", json=data).raise_for_status()
+
+    client.post(
+        f"/metadata/{guid}/aliases", json={"aliases": aliases}
+    ).raise_for_status()
+
+    try:
+        response = client.put(
+            f"/metadata/{guid}/aliases?merge={merge}", json={"aliases": updates}
+        )
+        assert str(response.status_code).startswith("20")
+
+        if merge:
+            excepted_aliases = sorted(list(set(aliases) | set(updates)))
```

Comment:
doesn't "excepted" mean the opposite of what this is? it might just be my english 👀

Reply:
whoops, that's supposed to be `expected_aliases` :man_facepalming: That's definitely my bad